### ISSUE TITLE
Incorrect fragment positions in Annotator "create" request

### ISSUE THREAD

**Issue by bug-rancher** [providedh/collaborative-platform#57]
When user try to add a second tag to tagged fragment, fragment positions are incorrect.
**This bug occur on Firefox, but not on Chromium.**

**How to replicate bug:**
**On branch "reference_to_element_of_the_list"**
Tested on "recipe_0.xml" file: https://raw.githubusercontent.com/providedh/ACDH_Salzburg_recipes/master/outputs/recipe_0.xml

1. Add first annotation to "Guetten" fragment:
```
"tag_name": "Ingredient"
"attribute_name": "ref"
"asserted_value": "object_recipe_0_xml-1"
```

Posisions in this request pointing to "Guetten" fragment, so it's ok

2. Add second annotation to annotated "Guetten" fragment:
```
"tag_name": "Ingredient"
"attribute_name": "ref"
"asserted_value": object_recipe_0_xml-2
```

Positions in this request pointing to "objectName ref="#object_recipe_0_xml-1">Guetten</objectName>". **Selected fragment can't start or end in the middle of the tag.**

**Comment by Janchorizo** [comment]
I have tried replicating this issue and the following is what I found:

After loading the adhered document and annotating it, there are some cases in which the 
annotation failed on me.

![image](https://user-images.githubusercontent.com/28607713/73086779-8cd3d000-3ed1-11ea-98af-bb8b7b0567fd.png)

Having those annotations, and selecting the fourth word in the paragraph (_Coppaun_), I
 the server fails at creating the annotation. The following screenshot shows the retrieved document
at app load, the message sent to the server, and the slice of the original document using 
the sent positions:

![image](https://user-images.githubusercontent.com/28607713/73086276-a32d5c00-3ed0-11ea-8b3d-b1238a8a067d.png)

The server-side code fails in the __add_tag_ method. 
https://github.com/providedh/collaborative-platform/blob/4f8706744ff6bfbc26f3f93368552a6a1815e024/src/collaborative_platform/apps/close_reading/annotator.py#L529-L630
The _annotated_fragment_ argument is
set to a different value than what is obtained in the front-end: `</date> Coppaun<gap/>`

**Comment by bug-rancher** [comment]
I replicated your bug. The root of this bug is this same as in first message in this issue - wrong positions in request. The difference is that positions pointing to fragment with part of the tag on the end instead on the beginning ('Coppaun<' instead 'objectName ref="#object_recipe_0_xml-1">Guetten' as in first post).

I added a validation to check for that exceptions, and now, if this exception will occur, backend will return a proper message with included fragment of xml based on positions from request. This will stop WebSockets from disconnecting in this cases and I hope this will help you to figure out why positions in request sometimes are wrong.

![Zrzut ekranu z 2020-01-28 13-45-44](https://user-images.githubusercontent.com/42303256/73265087-88aaf980-41d4-11ea-9ea6-be461bacea8b.png)
